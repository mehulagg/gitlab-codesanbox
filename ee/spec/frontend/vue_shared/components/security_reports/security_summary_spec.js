import { mount } from '@vue/test-utils';
import SecuritySummary from 'ee/vue_shared/security_reports/components/security_summary.vue';

describe('Severity Summary', () => {
  let wrapper;

  const createWrapper = message => {
    wrapper = mount({
      components: {
        SecuritySummary,
      },
      data() {
        return {
          message,
        };
      },
      template: `<div><security-summary :message="message" /></div>`,
    });
  };

  afterEach(() => {
    wrapper.destroy();
    wrapper = null;
  });

  describe.each([
    '',
    'foo',
    '%{criticalStart}1 critical%{criticalEnd}',
    '%{highStart}1 high%{highEnd}',
    '%{criticalStart}1 critical%{criticalEnd} and %{highStart}2 high%{highEnd}',
  ])('given the message %p', message => {
    beforeEach(() => {
      createWrapper(message);
    });

    it('interpolates correctly', () => {
      expect(wrapper.element).toMatchSnapshot();
    });
  });
});
