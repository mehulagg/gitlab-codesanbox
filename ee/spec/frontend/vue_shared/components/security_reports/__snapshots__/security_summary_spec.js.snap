// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Severity Summary given the message "" interpolates correctly 1`] = `<div />`;

exports[`Severity Summary given the message "%{criticalStart}1 critical%{criticalEnd} and %{highStart}2 high%{highEnd}" interpolates correctly 1`] = `
<div>
  <strong
    class="text-danger-800"
  >
    1 critical
  </strong>
   and 
  <strong
    class="text-danger-600"
  >
    2 high
  </strong>
</div>
`;

exports[`Severity Summary given the message "%{criticalStart}1 critical%{criticalEnd}" interpolates correctly 1`] = `
<div>
  <strong
    class="text-danger-800"
  >
    1 critical
  </strong>
</div>
`;

exports[`Severity Summary given the message "%{highStart}1 high%{highEnd}" interpolates correctly 1`] = `
<div>
  <strong
    class="text-danger-600"
  >
    1 high
  </strong>
</div>
`;

exports[`Severity Summary given the message "foo" interpolates correctly 1`] = `
<div>
  foo
</div>
`;
